packages/core/test/lib/base.test.ts

```diff
@@ -2014,4 +2014,59 @@ describe('BaseClient', () => {
       });
     });
   });
+
+  describe('hook removal with `on`', () => {
+    it('should return a cleanup function that, when executed, unregisters a hook', async () => {
+      expect.assertions(8);
+
+      const client = new TestClient(
+        getDefaultTestClientOptions({
+          dsn: PUBLIC_DSN,
+          enableSend: true,
+        }),
+      );
+
+      // @ts-expect-error Accessing private transport API
+      const mockSend = jest.spyOn(client._transport, 'send').mockImplementation(() => {
```

Comment:
```suggestion
      const mockSend = jest.spyOn(client['_transport'], 'send').mockImplementation(() => {
```

This way we do not need to ts-export-error this, I believe! :)